onnxruntime/core/platform/windows/dll_load_error.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <windows.h>
+#include <dbghelp.h>
+#include <iostream>
+#include <string>
+#pragma comment(lib, "dbghelp.lib")
+
+struct HMODULE_Deleter {
+  typedef HMODULE pointer;
+  void operator()(HMODULE h) { FreeLibrary(h); }
+};
+
+using ModulePtr = std::unique_ptr<HMODULE, HMODULE_Deleter>;
+
+// If a DLL fails to load, this will try loading the DLL and then its dependencies recursively
+// until it finds a missing file, then will report which file is missing and what the dependency
+// chain is.
+std::wstring DetermineLoadLibraryError(const wchar_t* filename_in) {
+  std::wstring error(L"Error loading");
+
+  std::wstring filename{filename_in};
+  while (filename.size()) {
+    error += std::wstring(L" \"") + filename + L"\"";
+
+    ModulePtr hModule = ModulePtr{LoadLibraryExW(filename.c_str(), NULL, DONT_RESOLVE_DLL_REFERENCES)};
+    if (!hModule) {
+      error += L" which is missing.";
+      return error;
+    }
+
+    // Get the address of the Import Directory
+    ULONG size;
+    PIMAGE_IMPORT_DESCRIPTOR importDesc = (PIMAGE_IMPORT_DESCRIPTOR)ImageDirectoryEntryToData(hModule.get(), TRUE, IMAGE_DIRECTORY_ENTRY_IMPORT, &size);
+    if (!importDesc) {
+      error += L" No import directory found.";  // This is unexpected, and I'm not sure how it could happen but we handle it just in case.
+      return error;
+    }
+
+    // Iterate through the import descriptors to see which dependent DLL can't load
+    filename.clear();
+    for (; importDesc->Characteristics; importDesc++) {
```

Comment:
Use `importDesc->Name != 0` instead of `importDesc->Characteristics`

Reply:
The header file disagrees:

```
typedef struct _IMAGE_IMPORT_DESCRIPTOR {
    union {
        DWORD   Characteristics;            // 0 for terminating null import descriptor
        DWORD   OriginalFirstThunk;         // RVA to original unbound IAT (PIMAGE_THUNK_DATA)
    } DUMMYUNIONNAME;

```

Where did you see for it to use Name instead?

Reply:
I saw some example code previously:
https://github.com/ruby/ruby/blob/35920f7a440139563e9e46686378bba062bd08c1/dln.c#L244

Reply:
Yeah, in the debugger the entire struct is all 0's, but the header file comment says to use the first DWORD.